client-mr/core/src/test/java/org/apache/hadoop/mapreduce/task/reduce/RssRemoteMergeManagerTest.java

```diff
@@ -58,22 +63,37 @@ public class RssRemoteMergeManagerTest {
   TaskAttemptID mapId2 = new TaskAttemptID(new TaskID(jobId, TaskType.MAP, 2), 0);
   TaskAttemptID reduceId1 = new TaskAttemptID(new TaskID(jobId, TaskType.REDUCE, 0), 0);
 
+  private static FileSystem remoteFS;
+  private static MiniDFSCluster cluster;
+
+  @BeforeAll
+  public static void setUpHdfs(@TempDir File tempDir) throws Exception {
+    Configuration conf = new Configuration();
+    File baseDir = tempDir;
+    conf.set(MiniDFSCluster.HDFS_MINIDFS_BASEDIR, baseDir.getAbsolutePath());
+    cluster = (new MiniDFSCluster.Builder(conf)).build();
+    String hdfsUri = cluster.getURI().toString() + "/";
+    remoteFS = (new Path(hdfsUri)).getFileSystem(conf);
+  }
+
+  @AfterAll
+  public static void tearDownHdfs() throws Exception {
+    remoteFS.close();
+    cluster.shutdown();
+  }
+
   @Test
-  public void mergerTest(@TempDir File tmpDir) throws Throwable {
```

Comment:
Why do we remove this code?

Reply:
> Why do we remove this code?

RssRemoteMergeManagerImpl should use remote filesystem. But in this unit test, we use local filesystem before. I change it, that's more reasonable.

Reply:
OK.

Reply:
Could you add some comments otherwise other people will change it TempDir, again.